packages/protocol/contracts/ACE/ACE.sol

```diff
@@ -1,87 +1,132 @@
 pragma solidity >=0.5.0 <0.6.0;
 
-import "./NoteRegistry.sol";
-import "../utils/NoteUtils.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import "../interfaces/IAZTEC.sol";
+
+import "../libs/IntegerUtils.sol";
+import "../libs/NoteUtils.sol";
+import "../libs/ProofUtils.sol";
+import "../libs/SafeMaths.sol";
 
 /**
  * @title The AZTEC Cryptography Engine
  * @author AZTEC
  * @dev ACE validates the AZTEC protocol's family of zero-knowledge proofs, which enables
- * digital asset builders to construct fungible confidential digital assets according to the AZTEC token standard.
+ *      digital asset builders to construct fungible confidential digital assets according to the AZTEC token standard.
  **/
-contract ACE {
-    // the commonReferenceString contains one G1 group element and one G2 group element,
+contract ACE is IAZTEC {
+    using IntegerUtils for uint256;
+    using NoteUtils for bytes;
+    using ProofUtils for uint24;
+    using SafeMath for uint256;
+    using SafeMath8 for uint8;
+
+    // bytes32 constant internal ACE_NOTE_SIGNATURE_TYPE_HASH = keccak256(abi.encodePacked(
```

Comment:
should delete commented code and replace with comment describing the string being hashed

Reply:
Fixed in https://github.com/AztecProtocol/AZTEC/pull/75